### DISCUSSION TITLE
Why the _document.js file should set in the src/pages folder?

### DISCUSSION THREAD

**Discussion by as17744** [vercel/next.js#21111] [Help]
I read the document and find out the  `_document.js ` was designed to set in the pages folder. But in my opinion the pages folder is used to set react component level pages. Does it more reasonable to set it under the root folder?

**Comment by bobarros** [comment]
I wonder if this should be in ideas and not help.

About your idea, the file _document is a react component that extends another one and is working fine for me and a lot of people =D

Don't you think that if you want to change something like this, it would be nice to offer some pros and cons?
Every deep discussion helps begginers like myself when we have the chance to read all the details. =D

**Reply by as17744** [reply]
I feel like it's not an idea but just my little wonder lol
